frame/pools/src/lib.rs

```diff
@@ -0,0 +1,1135 @@
+//! # Delegation pools for nominating
+//!
+//! This pallet allows delegators to delegate their stake to nominating pools, each of which acts as
+//! a nominator and nominates validators on their behalf.
+//!
+//! ## Design
+//!
+//! _Notes_: this section uses pseudo code to explain general design and does not necessarily
+//! reflect the exact implementation. Additionally, a strong knowledge of `pallet-staking`'s api is
+//! assumed.
+//!
+//! The delegation pool abstraction is composed of:
+//!
+//!  bonded pool: Tracks the distribution of actively staked funds. See [`BondedPool`] and
+//! [`BondedPools`]
+//! * reward pool: Tracks rewards earned by actively staked funds. See [`RewardPool`] and
+//!   [`RewardPooks`].
+//! * unbonding sub pools: This a collection of pools at different phases of the unbonding
+//!   lifecycle. See [`SubPools`] and [`SubPoolsStorage`].
+//! * delegators: Accounts that are members of pools. See [`Delegator`] and [`Delegators`].
+//! In order to maintain scalability, all operations are independent of the number of delegators. To
+//! do this, we store delegation specific information local to the delegator while the pool data
+//! structures have bounded datum .
+//!
+//! ### Design goals
```

Comment:
A common pattern in quality document-writing is to: 

```
//! You write the external docs here. The audience of this will be end-user. 
// You write the "implementation notes" here, the audience of this will be future maintainers of this code. 

Same patter can be applied to docs for a single struct as well:

```

/// External docs
pub struct Foo; 
impl Foo {
  // General design of Foo is explained here.
}
```

It seems like this `Design goals` is more of an implementation detail than a document. 

(feel free to ignore, bounding ideas around)